consensus/src/transaction_filter/mod.rs

```diff
@@ -14,19 +15,18 @@ impl TransactionFilter {
         Self { filter }
     }
 
-    pub fn filter(
-        &self,
-        block_id: HashValue,
-        timestamp: u64,
-        txns: Vec<SignedTransaction>,
-    ) -> Vec<SignedTransaction> {
+    pub fn filter(&self, block_id: HashValue, timestamp: u64, txns: &mut Vec<SignedTransaction>) {
         // Special case for no filter to avoid unnecessary iteration through all transactions in the default case
+        let num_txns_before = txns.len();
+        txns.retain(|txn| {
+            Duration::from_micros(timestamp).as_secs() < txn.expiration_timestamp_secs()
```

Comment:
have you checked to edge case - if txn has same expiration as block timestamp, are we skipping it right now?

Reply:
Yes. I added a test case for that. We are expiring those transactions now.